engines/efh/efh.cpp

```diff
@@ -1880,11 +1898,17 @@ bool EfhEngine::handleTalk(int16 monsterId, int16 arg2, int16 itemId) {
 						displayFctFullScreen();
 				}
 				setTextColorRed();
-				Common::KeyCode input = waitForKey();
-				if (input == Common::KEYCODE_y) {
+
+				setKeymap(kKeymapMenu);
+
+				waitForKey();
```

Comment:
Same here, make `waitForKey()` also returning `_customAction`, so you could write `if (input == kActionYes)`. The current code is confusing.